### DISCUSSION TITLE
Consider swapping V8 with Sobek (recent fork of Goja)

### DISCUSSION THREAD

**Discussion by nickchomey** [synadia-io/nex#369] [Nexlets and SDKS]
V8 makes for a heavy binary, etc... It would be neat to have a go-native implementation of JS. 

Goja seems to be the most popular option for this, but the k6 team at Grafana recently forked Goja into [Sobek](https://github.com/grafana/sobek?tab=readme-ov-file), where they merged their [long-outstanding PR](https://github.com/dop251/goja/pull/430#issuecomment-2288567164) for ES Module support. It also looks like they've been merging all other commits to Goja. 

Given these advancements, and the deeper resources available at Grafana, it seems to me that Sobek would be worth testing out for NEX's JS engine.

**Comment by jordan-rash** [comment]
We are actively experimenting with this now.  Will try and update when we have reached a decision on path forward.  Agree that V8 is heavy

**Comment by nickchomey** [comment]
As it turns out, sobek might not be appropriate (or at least would be difficult to implement). See this discussion I had with the sobek dev about limitations - both technical and license-based.

https://github.com/grafana/sobek/issues/49

**Reply by jordan-rash** [reply]
Thanks for sharing this @nickchomey
